### PR TITLE
Fix `abs` for complex arrays

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Towards: https://github.com/lfortran/lfortran/issues/3160

### CODE REVIEW

File: src/libasr/pass/intrinsic_function_registry.h [hunk 1]
```diff
@@ -1445,19 +1453,56 @@ namespace Abs {
...
+            double im;
+            double result;
+            ASR::expr_t *value_expr;
+            ASR::ArrayConstant_t * res;
+            for (size_t i = 0; i < arr->n_args; i++) {
+                value_expr = ASR::down_cast<ASR::ComplexConstructor_t>(arr->m_args)->m_value;
+                std::cout<<"this1"<<'\n';
+                re = ASR::down_cast<ASR::ComplexConstant_t>(value_expr)->m_re;
+                std::cout<<"re: "<<re<<'\n';
+                im = ASR::down_cast<ASR::ComplexConstant_t>(value_expr)->m_im;
```
Comment (HarshitaKalani) [inline src/libasr/pass/intrinsic_function_registry.h:1471]: Here, I wish to extract the complex elements from the array one by one and then create an array of reals and return it.
Can somebody please guide me on how can I access the complex elements here?

File: src/libasr/pass/array_op.cpp [hunk 2]
```diff
@@ -1270,8 +1270,29 @@ class ReplaceArrayOp: public ASR::BaseExprReplacer<ReplaceArrayOp> {
...
+            std::cout<<"hk1"<<'\n';
+            // ASR::Function_t *func = ASR::down_cast<ASR::Function_t>(ASRUtils::symbol_get_past_external(x->m_type));
+            ASR::Function_t *func = ASR::down_cast<ASR::Function_t>(x->m_type);
+
+            std::cout<<"hk2"<<'\n';
+            if(func->m_return_var != nullptr) {
+                std::cout<<"yaha1"<<'\n';
+            }
+            
+            if (func->m_return_var != nullptr && !ASRUtils::is_array(ASRUtils::expr_type(func->m_return_var))) {
```
Comment (HarshitaKalani) [inline src/libasr/pass/array_op.cpp:1282]: Can somebody suggest how can I deal with `T *x`? I wish to check if it is an array.

Comment (HarshitaKalani) [reply src/libasr/pass/array_op.cpp:1282]: @Pranavchiku I believe this is quite similar to one of the issues you resolved. Can you please advise?

Comment (Pranavchiku) [reply src/libasr/pass/array_op.cpp:1282]: Looking into it.

**Comment by Pranavchiku** [comment]
Closing with reference to #3197.

**Pull request closed without merging**